scipy/optimize/tests/test_minpack.py

```diff
@@ -539,6 +540,24 @@ def test_NaN_handling(self):
         assert_raises(ValueError, curve_fit, lambda x, a, b: a*x + b,
                       xdata, ydata, **{"check_finite": True})
 
+    def test_ignore_nan(self):
+        # Test for handling of NaNs in input data when ignore_nan flag is set: gh-11841
+        f = lambda x, a, b: a*x + b
+
+        xdata = np.array([1, 2, np.nan, 3, 4, np.nan])
+        ydata = np.array([3, 4, 5, 6, np.nan, 7])
+        try:
+            curve_fit(f, xdata, ydata, ignore_nan=True)
+        except ValueError:
```

Comment:
This construct is unnecessary. If a `ValueError` is raised, the test will simply fail anyway which is fine. Can you just do
```
res = curve_fit(f, xdata, ydata, ignore_nan=True)
# then assert `res` contains the expected result
```

Reply:
Thanks!
I finished fixing.